src/envs/pybullet_robots.py

```diff
@@ -129,6 +132,13 @@ def set_motors(self, action_arr: Array) -> None:
         """Update the motors to execute the given action in PyBullet."""
         raise NotImplementedError("Override me!")
 
+    @abc.abstractmethod
+    def run_inverse_kinematics(self,
+                               end_effector_pose: Pose3D,
+                               validate: bool = True) -> List[float]:
```

Comment:
do we really want validate = True as a default arg? aren't we using it as False? do we ever use it as True? maybe make it a non-option arg so people have to think about it?